### PR TITLE
[zh] Rework Deployment concept

### PR DESCRIPTION
**Pull request by tengqm**
*** WARNING ***
This is a huge page. It takes some time to review.
There are quite some translations that prove the original author didn't understand what the doc is talking about.

### CODE REVIEW

**Comment by sftim** [comment]
Oops, wrong PR.

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 1]
```diff
@@ -1,512 +1,499 @@
...
 <!--
  * A Deployment named `nginx-deployment` is created, indicated by the `.metadata.name` field.
--->
-* 将创建名为 `nginx-deployment` 的 Deployment ，由 `.metadata.name` 字段指示。
-
-<!--
  * The Deployment creates three replicated Pods, indicated by the `replicas` field.
 -->
-*  Deployment 创建三个复制的 Pods，由 `replicas` 字段指示。
+* 创建名为 `nginx-deployment`（由 `.metadata.name` 字段标明）的 Deployment，。
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:100]:
```suggestion
* 创建名为 `nginx-deployment`（由 `.metadata.name` 字段标明）的 Deployment。
```

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 2]
```diff
@@ -1,512 +1,499 @@
...
-    and the values array contains only "value".
-    All of the requirements, from both `matchLabels` and `matchExpressions`, must be satisfied in order to match.
+  The `matchLabels` field is a map of {key,value} pairs. A single {key,value} in the `matchLabels` map
+  is equivalent to an element of `matchExpressions`, whose key field is "key" the operator is "In",
+  and the values array contains only "value".
+  All of the requirements, from both `matchLabels` and `matchExpressions`, must be satisfied in order to match.
   -->
-    `matchLabels` 字段是 {key,value} 的映射。单个 {key,value}在 `matchLabels` 映射中的值等效于 `matchExpressions` 的元素，其键字段是“key”，运算符为“In”，值数组仅包含“value”。所有要求，从 `matchLabels` 和 `matchExpressions`，必须满足才能匹配。
+  {{< note >}}
+  `matchLabels` 字段是 `{key,value}` 耦对的映射。在 `matchLabels` 映射中的单个 {key,value}
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:120]: 耦对？

Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:120]: {key,value} -> `{key,value}`

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 3]
```diff
@@ -1,512 +1,499 @@
...
+  映射等效于 `matchExpressions` 中的一个元素，即其 key 字段是 “key”，operator 为 “In”，value
+  数组仅包含“value”。在 `matchLabels` 和 `matchExpressions` 中给出的所有条件都必须满足才能匹配。
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:122]:
```suggestion
  数组仅包含 “value”。在 `matchLabels` 和 `matchExpressions` 中给出的所有条件都必须满足才能匹配。
```

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 4]
```diff
@@ -1,512 +1,499 @@
...
-  * 创建一个容器并使用`name`字段将其命名为 `nginx`。
-
-<!--
- Follow the steps given below to create the above Deployment:
+    [Docker Hub](https://hub.docker.com/) image at version 1.14.2.
+  * Create one container and name it `nginx` using the `name` field.
 -->
-按照以下步骤创建上述 Deployment ：
+* `template` 字段包含以下子字段：
+  * Pod 被使用`labels`字段打上 `app: nginx` 标签。
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:134]:
```suggestion
  * Pod 被使用 `labels` 字段打上 `app: nginx` 标签。
```

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 5]
```diff
@@ -1,512 +1,499 @@
...
-    * `UP-TO-DATE`显示已更新以实现期望状态的副本数。
-    * `AVAILABLE`显示应用程序可供用户使用的副本数。
-    * `AGE` 显示应用程序运行的时间量。
-
-    <!--
-    Notice how the number of desired replicas is 3 according to `.spec.replicas` field.
-    -->
-    请注意，根据`.spec.replicas`副本字段，所需副本的数量为 3。
+2. 运行 `kubectl get deployments` 检查 Deployment 是否已创建。如果仍在创建 Deployment，
+   则输出与下面的例子类似：
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:168]:
```suggestion
   输出类似于：
```

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 6]
```diff
@@ -1,512 +1,499 @@
...
+   * `CURRENT` displays how many replicas are currently running.
+   * `READY` displays how many replicas of the application are available to your users.
+   * `AGE` displays the amount of time that the application has been running.
+   -->
+   ReplicaSet 输出中包含以下字段：
+
+   * `NAME` 列出名字空间中 ReplicaSet 的名称；
+   * `DESIRED` 显示应用的期望副本个数，即在创建 Deployment 时所定义的值。
+     此为期望状态；
+   * `CURRENT` 显示当前运行状态中的副本个数；displays how many replicas are currently running.
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:250]: displays how many replicas are currently running.   未翻译

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 7]
```diff
@@ -545,623 +532,628 @@ up to 3 replicas, as well as scaling down the old ReplicaSet to 0 replicas.
...
+  ```shell
+  kubectl rollout status deployment.v1.apps/nginx-deployment
+  ```
 
-    <!--
-    The output is similar to this:
-    -->
-    输出：
+  <!-- The output is similar to this: -->
+  输出：
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:683]: 输出类似于：

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 8]
```diff
@@ -545,623 +532,628 @@ up to 3 replicas, as well as scaling down the old ReplicaSet to 0 replicas.
...
-        * Append the `--record` flag to save the `kubectl` command that is making changes to the resource.
-        * Manually editing the manifest of the resource.
-    -->
-    * 使用 `kubectl annotate deployment.v1.apps/nginx-deployment kubernetes.io/change-cause="image updated to 1.9.1"`  Deployment 对 Deployment 进行分号。
-    * 追加 `--record` 以保存正在更改资源的 `kubectl` 命令。
-    * 手动编辑资源的清单。
+   <!--
+   `CHANGE-CAUSE` is copied from the Deployment annotation `kubernetes.io/change-cause` to its revisions upon creation. You can specify the`CHANGE-CAUSE` message by:
+   -->
+   `CHANGE-CAUSE` 的内容时从 Deployment 的 `kubernetes.io/change-cause` 注解复制过来的。
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:835]: 时？

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 9]
```diff
@@ -545,623 +532,628 @@ up to 3 replicas, as well as scaling down the old ReplicaSet to 0 replicas.
...
 
 <!--
- Assuming [horizontal Pod autoscaling](/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/) is enabled
+Assuming [horizontal Pod autoscaling](/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/) is enabled
 in your cluster, you can setup an autoscaler for your Deployment and choose the minimum and maximum number of
 Pods you want to run based on the CPU utilization of your existing Pods.
 -->
-假设启用[水平自动缩放 Pod](/zh/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/)在集群中，可以为 Deployment 设置自动缩放器，并选择最小和最大
-要基于现有 Pods 的 CPU 利用率运行的 Pods。
+假设集群启用了[Pod 的水平自动缩放 Pod](/zh/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/)，
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:1025]: Pod 的水平自动缩放 Pod？

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 10]
```diff
@@ -545,623 +532,628 @@ up to 3 replicas, as well as scaling down the old ReplicaSet to 0 replicas.
...
+你可以为 Deployment 设置自动缩放器，并基于现有 Pods 的 CPU 利用率选择
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:1026]: 选择选择？

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 11]
```diff
@@ -1170,306 +1162,280 @@ nginx-deployment-618515232    11        11        11        7m
...
+  ```
 
-{{< note >}}
 <!--
- You cannot rollback a paused Deployment until you resume it.
+You cannot rollback a paused Deployment until you resume it.
 -->
-暂停的 Deployment 不可以回滚，除非恢复它以后。
+{{< note >}}
+你不可以会滚处于暂停状态的 Deployment，除非先恢复其执行状态。
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:1366]: 会滚？

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 12]
```diff
@@ -1170,306 +1162,280 @@ nginx-deployment-618515232    11        11        11        7m
...
 -->
-Kubernetes 将 Deployment 标记为 _完成_，当它具有以下特征时：
+### 完成的 Deployment    {#complete-deployment}
 
-<!--
- * All of the replicas associated with the Deployment have been updated to the latest version you've specified, meaning any
-updates you've requested have been completed.
--->
-* 与 Deployment 关联的所有副本都已更新到指定的最新版本，这意味着请求的任何更新都已完成。
+当 Deployment 具有以下特征时，Kubernetes 将其标记为 _完成（Complete）_，：
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:1414]:
```suggestion
当 Deployment 具有以下特征时，Kubernetes 将其标记为 _完成（Complete）_ ：
```

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 13]
```diff
@@ -1479,15 +1445,16 @@ $ echo $?
...
-###  Deployment 失败
+### 失败的 Deployment   {#failed-deployment}
 
 <!--
- Your Deployment may get stuck trying to deploy its newest ReplicaSet without ever completing. This can occur
+Your Deployment may get stuck trying to deploy its newest ReplicaSet without ever completing. This can occur
 due to some of the following factors:
 -->
-你的 Deployment 可能会在未完成的情况下尝试 Deployment 其最新的 ReplicaSet 时遇到问题。可能发生此情况由于以下一些因素：
+你的 Deployment 可能会在尝试部署其最新的 ReplicaSet 受挫，一直处于未完成状态。
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:1456]: 受挫  ?   失败？

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 14]
```diff
@@ -1656,347 +1629,360 @@ Conditions:
...
 -->
-`.spec.strategy` 策略指定用于用新 Pods 替换旧 Pods 的策略。`.spec.strategy.type` 可以是“Recreate”或“RollingUpdate”。“RollingUpdate”是默认值。
+### 策略   {#strategy}
 
-<!--
- #### Recreate Deployment
--->
-#### 重新创建 Deployment
+`.spec.strategy` 策略指定用于用新 Pods 替换旧 Pods 的策略。
+`.spec.strategy.type` 可以是“Recreate”或“RollingUpdate”。“RollingUpdate”是默认值。
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:1836]:
```suggestion
`.spec.strategy.type` 可以是 “Recreate” 或 “RollingUpdate”。“RollingUpdate” 是默认值。
```

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 15]
```diff
@@ -1656,347 +1629,360 @@ Conditions:
...
-`.spec.minReadySeconds` 是一个可选字段，用于指定新创建的 Pod 在没有任意容器崩溃情况下的最小就绪时间，以便将其视为可用。默认值为 0（Pod 在准备就绪后立即将被视为可用）。了解有关何时Pod 被视为已准备就绪，参考[容器探针](/zh/docs/concepts/workloads/pods/pod-lifecycle/#container-probes)。
+### 最短就绪时间    {#min-ready-seconds}
 
-<!--
- ### Rollback To
--->
-### 回滚
+`.spec.minReadySeconds` 是一个可选字段，用于指定新创建的 Pod 在没有任意容器崩溃情况下的最小就绪时间，
+只有超出这个时间 Pod 才被视为可用。默认值为 0（Pod 在准备就绪后立即将被视为可用）。
+如欲了解何时 Pod 被视为就绪，可参考[容器探针](/zh/docs/concepts/workloads/pods/pod-lifecycle/#container-probes)。
```
Comment (tanjunchen) [inline content/zh/docs/concepts/workloads/controllers/deployment.md:1947]: 如欲？

File: content/zh/docs/concepts/workloads/controllers/deployment.md [hunk 16]
```diff
@@ -1,512 +1,499 @@
...
 
 <!--
-
 You describe a _desired state_ in a Deployment, and the Deployment controller changes the actual state to the desired state at a controlled rate. You can define Deployments to create new ReplicaSets, or to remove existing Deployments and adopt all their resources with new Deployments.
 
 -->
-描述 Deployment 中的 _desired state_，并且 Deployment 控制器以受控速率更改实际状态，以达到期望状态。可以定义 Deployments 以创建新的 ReplicaSets ，或删除现有 Deployments ，并通过新的 Deployments 使用其所有资源。
+你负责描述 Deployment 中的 _目标状态_，而 Deployment 控制器以受控速率更改实际状态，
+使其变为期望状态。你可以定义 Deployment 以创建新的 ReplicaSet，或删除现有 Deployment，
+并通过新的 Deployment 收养其资源。
```
[the remark this answers is not in the document]
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:37]: 是的。形象说法。Orphaned Pods gets adopted by the new Deployment.

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 1 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:100]: whoops, fixed.

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 2 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:120]: 拼音输入有误，应为”偶对“。Fixed。

Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:120]: Done

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (tanjunchen) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:37]: right

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 5 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:168]: hoho，makes sense. 现在的说法有点儿生硬，是”similar to the following" 的直译。

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 7 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:683]: OK

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 6 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:250]: 翻译了，只是忘记删除这个英文原文。已修改。

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 8 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:835]: done

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 10 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:1026]: fixed.

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 11 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:1366]: Mac 内置的中文输入法让人很无语。

Comment (tanjunchen) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:1366]: haha

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 13 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:1456]: 受挫是较人性化的说法，英文原文是 “stuck”，纠结了半天，“失败”也是不准确的。

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 14 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:1836]: 因为是弯引号，所以没有留白。应该问题不大。

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 15 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:1947]: 好吧，太文绉绉了是吧。

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 9 above)
Comment (tengqm) [reply content/zh/docs/concepts/workloads/controllers/deployment.md:1025]: fixed.

File: content/zh/docs/concepts/workloads/controllers/deployment.md (hunk 15 above)
Comment (tanjunchen) [reply content/zh/docs/concepts/workloads/controllers/deployment.md]: 有点。还是简单点好。

**Comment by tanjunchen** [comment]
/lgtm
/cc @xichengliudui 
for approve

**Pull request merged**